builtin/logical/aws/secret_access_keys.go

```diff
@@ -37,6 +42,75 @@ func secretAccessKeys(b *backend) *framework.Secret {
 	}
 }
 
+func (b *backend) secretAccessKeysAndTokenCreate(s logical.Storage,
+	displayName, policyName, policy string,
+	lifeTimeInSeconds *int64) (*logical.Response, error) {
+	IAMClient, err := clientIAM(s)
+	if err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	}
+	STSClient, err := clientSTS(s)
+	if err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	}
+
+	// Generate a random username. We don't put the policy names in the
+	// username because the AWS console makes it pretty easy to see that.
+	username := fmt.Sprintf("vault-%s-%d-%d", normalizeDisplayName(displayName), time.Now().Unix(), rand.Int31n(10000))
+
+	// Write to the WAL that this user will be created. We do this before
+	// the user is created because if switch the order then the WAL put
+	// can fail, which would put us in an awkward position: we have a user
+	// we need to rollback but can't put the WAL entry to do the rollback.
+	walId, err := framework.PutWAL(s, "user", &walUser{
+		UserName: username,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("Error writing WAL entry: %s", err)
+	}
+
+	// Create the user
+	_, err = IAMClient.CreateUser(&iam.CreateUserInput{
+		UserName: aws.String(username),
+	})
+	if err != nil {
+		return logical.ErrorResponse(fmt.Sprintf(
+			"Error creating IAM user: %s", err)), nil
+	}
+
+	// Create the keys and token
+	fmt.Println(username, policy)
```

Comment:
Is there a proper way to handle this logging? It looks quite ugly in the output of ``vault server -dev -log-level=trace``:

```
2016/01/14 08:06:12 [DEBUG] rollback: attempting rollback on secret/
2016/01/14 08:06:12 [DEBUG] rollback: attempting rollback on aws/
2016/01/14 08:06:12 [DEBUG] rollback: attempting rollback on cubbyhole/
2016/01/14 08:06:12 [DEBUG] rollback: attempting rollback on sys/
vault-root-1452776776-3632 {"Version":"2012-10-17","Statement":{"Effect":"Allow","Action":"*","Resource":"*"}}
2016/01/14 08:07:12 [DEBUG] rollback: attempting rollback on aws/
2016/01/14 08:07:12 [DEBUG] rollback: attempting rollback on secret/
2016/01/14 08:07:12 [DEBUG] rollback: attempting rollback on cubbyhole/
```

I don't know if Vault has a real logging subsystem, or if there's just a better way to format this with the date, and ideally with some indication of where it's coming from?